### PR TITLE
fix(amazonq): Fix memory report telemetry of Amazon Q LSP

### PR DESCRIPTION
**Pull request by leigaol**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
The memory usage from LSP API is in bytes, but we need to report MB.

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextProvider.kt [hunk 1]
```diff
@@ -162,7 +162,7 @@ class ProjectContextProvider(val project: Project, private val encoderServer: En
         logger.debug { "project context index time: ${duration}ms" }
         if (connection.responseCode == 200) {
             val usage = getUsage()
-            TelemetryHelper.recordIndexWorkspace(duration, filesResult.files.size, filesResult.fileSize, true, usage?.memoryUsage, usage?.cpuUsage, startUrl)
+            TelemetryHelper.recordIndexWorkspace(duration, filesResult.files.size, filesResult.fileSize, true, usage?.memoryUsage?.div(1024 * 1024), usage?.cpuUsage, startUrl)
```
Comment (zixlin7) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextProvider.kt:165]: will `div` break if it's 0?

File: .changes/next-release/bugfix-16f02728-142e-4020-b24d-eff66ecd17e9.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "type" : "bugfix",
+  "description" : "Fix language server memory usage telemetry"
```
Comment (Will-ShaoHua) [inline .changes/next-release/bugfix-16f02728-142e-4020-b24d-eff66ecd17e9.json:3]: it's not customer facing right?

Comment (Will-ShaoHua) [reply .changes/next-release/bugfix-16f02728-142e-4020-b24d-eff66ecd17e9.json:3]: dont need a changelog i think

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextProvider.kt (hunk 1 above)
Comment (Will-ShaoHua) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextProvider.kt:165]: noppe

**Review by Will-ShaoHua — APPROVED** [review]

**Comment by Will-ShaoHua** [comment]
```
eecution failed for task ':plugin-amazonq:chat:jetbrains-community:detekt'.
```

File: .changes/next-release/bugfix-16f02728-142e-4020-b24d-eff66ecd17e9.json (hunk 2 above)
Comment (zixlin7) [reply .changes/next-release/bugfix-16f02728-142e-4020-b24d-eff66ecd17e9.json:3]: +1

**Review by zixlin7 — APPROVED** [review]

**Review by rli — APPROVED** [review]

**Pull request merged**